From 2f6560f5afe7b8f136c31823af8b2d93ffa73163 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 17 May 2023 11:56:40 +0200
Subject: [PATCH 0370/1141] drivers: watchdog: arm_smc_wdt: revert
 WDOG_HW_RUNNING at probe

This reverts commit 6e345dfc6a50f8637959b33e9b4f55ce5f664e80.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ib7ee70857722962f4892fb86e0d3b49dd89f667a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/314006
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/watchdog/arm_smc_wdt.c | 9 ---------
 1 file changed, 9 deletions(-)

--- a/drivers/watchdog/arm_smc_wdt.c
+++ b/drivers/watchdog/arm_smc_wdt.c
@@ -148,15 +148,6 @@ static int smcwd_probe(struct platform_d
 	if (err)
 		return err;
 
-	if (IS_ENABLED(CONFIG_WATCHDOG_HANDLE_BOOT_ENABLED)) {
-		err = smcwd_start(wdd);
-		if (err)
-			return err;
-
-		/* Make sure the watchdog is serviced */
-		set_bit(WDOG_HW_RUNNING, &wdd->status);
-	}
-
 	err = devm_watchdog_register_device(&pdev->dev, wdd);
 	if (err)
 		return err;
